edgelet/edgelet-http/src/util/hyperwrap.rs

```diff
@@ -36,17 +51,41 @@ impl Config {
         if self.null {
             Ok(Client::Null)
         } else {
-            let config = self.clone();
+            let mut builder = TlsConnector::builder();
+            if let Some(bundle) = &self.trust_bundle {
+                let certs = X509::stack_from_pem(&bundle.cert)
+                    .with_context(|_| ErrorKind::TrustBundle)?;
+                for cert in certs {
+                    let der = cert.to_der().with_context(|_| ErrorKind::TrustBundle)?;
+                    let c = TlsCertificate::from_der(&der)
+                        .with_context(|_| ErrorKind::TrustBundle)?;
+                    builder.add_root_certificate(c);
+                }
+            }
+            let connector = if let Some(id) = &self.identity_certificate {
+                let identity = id.get_identity()?;
+                builder
+                    .identity(identity)
+                    .build()
+                    .with_context(|_| ErrorKind::CertificateConversionError)
```

Comment:
Is this intentionally not `ErrorKind::Initialization` like the non-identity-cert case?

Otherwise this would be better written as:

```rust
if let Some(id) = &self.identity_certificate {
    let identity = id.get_identity()?;
    builder.identity(identity);
}

let connector = builder.build() ...
```

Reply:
Thanks fixed.